docs(feature gates): updates content due to maturity (#9521)
strimzi/strimzi-kafka-operator: documentation/modules/operators/ref-operator-cluster-feature-gates.adoc

// deploying/deploying.adoc

[id='ref-operator-cluster-feature-gates-{context}']
= Feature gates

[role="_abstract"]
Strimzi operators use feature gates to enable or disable specific features and functions.
Enabling a feature gate alters the behavior of the associated operator, introducing the corresponding feature to your Strimzi deployment.

The purpose of feature gates is to facilitate the trial and testing of a feature before it is fully adopted. 

> The preview state (enabled or disabled) of a feature gate may vary by default, depending on its maturity level.

Reviewer: I would avoid calling it a preview as it is not always the purpose of the feature gate to preview something. Often it is also to gradually introduce some changes.

```suggestion
The state (enabled or disabled) of a feature gate may vary by default, depending on its maturity level.
```

As a feature gate matures and reaches General Availability (GA), it transitions to an enabled state by default and becomes a permanent part of the Strimzi deployment.
A feature gate at the GA stage cannot be disabled. 

> == Mature feature gates (GA)

Reviewer 2: Maybe "graduated" instead of "mature"?

IMPORTANT: With the `UseStrimziPodSets` feature gate permanently enabled, direct downgrades from Strimzi 0.35 and newer to Strimzi 0.27 or earlier are not possible. You must first downgrade through one of the Strimzi versions in-between, disable the `UseStrimziPodSets` feature gate, and then downgrade to Strimzi 0.27 or earlier.

[id='ref-operator-stable-connect-identities-feature-gate-{context}']
=== StableConnectIdentities feature gate

The `StableConnectIdentities` feature gate introduced the `StrimziPodSet` custom resource for managing Kafka Connect and Kafka MirrorMaker 2 pods, replacing the use of Kubernetes `Deployment` resources.

`StrimziPodSet` resources give the pods stable names and stable addresses, which do not change during rolling upgrades, replacing the use of Kubernetes `Deployment` resources.

IMPORTANT: With the `StableConnectIdentities` feature gate permanently enabled, direct downgrades from Strimzi 0.39 and newer to Strimzi 0.33 or earlier are not possible.
You must first downgrade through one of the Strimzi versions in-between, disable the `StableConnectIdentities` feature gate, and then downgrade to Strimzi 0.33 or earlier.

> == Preview feature gates (alpha/beta)

Reviewer: As I said, I would not necessarily refer to all feature gates as _preview_. And especially not when they are at beta stage and enabled by default. This applies also in the text below.

> == Early access feature gates (alpha/beta)

Author: I think we need three sections then. I've split as follows, ordering by maturity:

Graduated feature gates (GA)
Stable feature gates (Beta)
Early access feature gates (Alpha)

> == Early access feature gates (alpha/beta)

Reviewer 2: Looks good, but I would remove the word gates. There is no gate when the feature is in GA stage.

Preview feature gates that have not yet reached the GA stage can be enabled or disabled. 

> A preview feature gate provides an opportunity for assessment before its functionality is permanently incorporated into AMQ Streams.

Reviewer 2: ```suggestion
A preview feature gate provides an opportunity for assessment before its functionality is permanently incorporated into Strimzi.
```

[id='ref-operator-use-kraft-feature-gate-{context}']
=== UseKRaft feature gate

The `UseKRaft` feature gate has a default state of _disabled_.

> The `UseKRaft` feature gate deploys a Kafka cluster in KRaft (Kafka Raft metadata) mode without ZooKeeper.

Reviewer 2: ```suggestion
The `UseKRaft` feature gate deploys a Kafka cluster in KRaft mode without ZooKeeper.
KRaft is a term that refers to the Kafka Raft dialect used for replicating the cluster metadata.
```

Reviewer: I would not change this. I think the original is more understandable to users and makes more sense in general. I do not think _Raft dialect_ has any real meaning.

Reviewer 2: It's shorter, but not more understandable IMO. The term "Raft dialect" was mentioned in the [original KIP](https://cwiki.apache.org/confluence/display/KAFKA/KIP-595%3A+A+Raft+Protocol+for+the+Metadata+Quorum).

Reviewer: Just because it is used in some KIP does not mean it is understandable. If nothing else, the audiences are very different.

Author: Raft dialect isn't used in the Kafka docs. I don't think we should start using it.

.Disabling the `UnidirectionalTopicOperator` feature gate

To disable the `UnidirectionalTopicOperator` feature gate, specify `-UnidirectionalTopicOperator` in the `STRIMZI_FEATURE_GATES` environment variable in the Cluster Operator configuration.

> == Enabling preview feature gates

Reviewer: ```suggestion
== Enabling feature gates
```
